package chaos;
import "gtfs-realtime.proto"

message Disruption{
    required string id = 1;
    //indicate if this disruption must be deleted
    //in this case all others fields will be null
    optional bool is_deleted = 2 [default = false];

    optional string reference = 3;

    // the publication period specify when an information can be displayed to
    // the customer, if a request is made before or after this period the
    // disruption must not be shown
    optional transit_realtime.TimeRange publication_periods = 4;

    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 created_at = 6;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 updated_at = 7;

    optional Cause cause = 8;

    repeated Impact impacts = 9;

}

message Cause{
    required string id = 1;
    optional string wording = 2;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 created_at = 3;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 updated_at = 4;
}

message Impact{
    required string id = 1;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 created_at = 2;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 updated_at = 3;

    // the application period define when the impact happen
    repeated transit_realtime.TimeRange application_periods = 4;

    optional Severity severity = 5;

    repeated PtObject pt_objects = 6;

    repeated Message messages = 6;
}

message Severity{
    enum Effect {
        NONE = 1;
        BLOCKED = 2;
    }
    required string id = 1;
    optional string wording = 2;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 created_at = 3;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 updated_at = 4;

    optional string color = 5;

    optional int32 priority = 6;

    optional Effect effect = 7;
}

message Message{
    required string text = 1;
    required channel Channel = 2;

    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 created_at = 3;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 updated_at = 4;
}

// Maybe we can use the EntitySelector of gtfs RT here
message PtObject{
    enum Type{
        NETWORK = 1;
        STOPAREA = 2;
    }
    required string uri = 1;
    required Type type = 2;
}

message Channel{
    required string id = 1;
    optional string name = 2
    optional string content_type = 3;

    // maximum number of character for this channel, mostly for information
    // since diffusion system don't need to bother with that
    // If missing, there is no limit!
    optional uint32 max_size = 4;

    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 created_at = 5;
    //In POSIX time (i.e., number of seconds since January 1st 1970 00:00:00 UTC).
    optional uint64 updated_at = 6;
}
